fairscale/nn/moe/moe_layer.py

```diff
@@ -55,13 +55,17 @@ class MOELayer(Base):
             expert network
     """
 
-    def __init__(self, gate: Module, expert: Module, group: Optional[Any] = None) -> None:
+    def __init__(self, gate: Module, experts: Union[Module, ModuleList], group: Optional[Any] = None) -> None:
         super().__init__()
         self.gate = gate
-        self.expert = expert
+        if type(experts) == ModuleList:
+            self.experts = cast(ModuleList, experts)
```

Comment:
The cast in this line is redundant, right? Why not use `self.experts = experts`?

Reply:
experts is of type Union[Module, ModuleList]. The type-checker cannot follow the control flow of the program so is unaware that in this cast experts is ModuleList.